### PR TITLE
feat(debug): Add databases=managed debug option

### PR DESCRIPTION
**Pull request by kyleconroy**
- Remove the --no-database flag
- Refactor the shfmt package

### CODE REVIEW

File: internal/engine/postgresql/analyzer/analyze.go [hunk 1]
```diff
@@ -204,8 +209,10 @@ func (a *Analyzer) Analyze(ctx context.Context, n ast.Node, query string, migrat
 				return nil, err
 			}
 			uri = edb.Uri
+		} else if a.dbg.OnlyManagedDatabases {
+			return nil, fmt.Errorf("database: connections disabled via command line flag")
```
Comment (andrewmbenton) [inline internal/engine/postgresql/analyzer/analyze.go:213]:
```suggestion
			return nil, fmt.Errorf("database: connections disabled via SQLCDEBUG=databases=managed")
```

File: internal/cmd/vet.go [hunk 2]
```diff
@@ -488,7 +481,7 @@ func (c *checker) checkSQL(ctx context.Context, s config.SQL) error {
 	var prep preparer
 	var expl explainer
 	if s.Database != nil { // TODO only set up a database connection if a rule evaluation requires it
-		if c.NoDatabase {
+		if s.Database.URI != "" && c.OnlyManagedDB {
 			return fmt.Errorf("database: connections disabled via command line flag")
```
Comment (andrewmbenton) [inline internal/cmd/vet.go:485]:
```suggestion
			return fmt.Errorf("database: connections disabled via SQLCDEBUG=databases=managed")
```

File: internal/cmd/vet.go [hunk 3]
```diff
@@ -448,14 +448,7 @@ func (c *checker) fetchDatabaseUri(ctx context.Context, s config.SQL) (string, f
 }
 
 func (c *checker) DSN(dsn string) (string, error) {
```
Comment (andrewmbenton) [inline internal/cmd/vet.go:450]: Don't really need this func anymore

**Review by andrewmbenton — CHANGES_REQUESTED** [review]

File: internal/cmd/vet.go (hunk 3 above)
Comment (kyleconroy) [reply internal/cmd/vet.go:450]: Agreed, but it was easier to keep it around

**Review by andrewmbenton — APPROVED** [review]

**Pull request merged**